src/main/java/com/venafi/vcert/sdk/certificate/CertificateRequest.java

```diff
@@ -282,25 +288,28 @@ public boolean checkCertificate(Certificate certificate) throws VCertException {
         PKCS10CertificationRequest csr =
             new PKCS10CertificationRequest(pemReader.readPemObject().getContent());
         pemReader.close();
+        AlgorithmNameFinder nameFinder = new DefaultAlgorithmNameFinder();
```

Comment:
Is that the best way to find the algorithm name?

Reply:
seems that is the one provided by the bouncycastle library. so for this case yes.